esp-hal/src/clock/clocks_ll/esp32.rs

```diff
@@ -183,9 +183,9 @@ pub(crate) fn esp32_rtc_update_to_xtal(freq: XtalClock, _div: u32) {
         });
 
         // adjust ref_tick
-        apb_cntl.xtal_tick_conf().as_ptr().write_volatile(
-            ((freq.hz()) / REF_CLK_FREQ - 1) | apb_cntl.xtal_tick_conf().as_ptr().read_volatile(),
```

Comment:
that was probably slightly incorrect before since it didn't mask out the previous value of xtal_tick_num